appc/appconnector.go

```diff
@@ -152,6 +227,18 @@ func (e *AppConnector) updateRoutes(routes []netip.Prefix) {
 
 	var toRemove []netip.Prefix
 
+	// If we're storing routes and know e.controlRoutes is a good
+	// representation of what should be in AdvertisedRoutes we can stop
+	// advertising routes that used to be in e.controlRoutes but are not
+	// in routes.
+	if e.shouldStoreRoutes {
+		for _, old := range e.controlRoutes {
+			if !slices.Contains(routes, old) {
```

Comment:
This is O(n^2) and should probably be replaced with something that can be looked up quickly (maybe netipx.IPset, route.Set, a sorted slice + binary search, or something else).